wgpu-hal/src/gles/egl.rs

```diff
@@ -757,13 +756,30 @@ impl crate::Surface<super::Api> for Surface {
             );
         }
 
-        let format_desc = device.shared.describe_texture_format(config.format);
         let gl = &device.shared.context;
+
+        // Check renderer version to see if we need to workaround intel mesa bug
+        if self.enable_intel_mesa_fastclear_workaround.is_none() {
```

Comment:
this needs to happen in the `expose` function, where the renderer string is analyzed. And the result flag should go into `PrivateCapabilities` ideally

Reply:
> this needs to happen in the expose function,

That's what I wanted to do initially, but a surface can be created without having enumerated adapters, so I didn't know the best way to go about getting to the capability information exposed by the adapter inside the surface, when the adapter could have been exposed at any time _after_ the surface was created.

Any ideas for how to do that?

Reply:
Would this still hold when #1656 is merged?